File: fix_get_package.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('tests/**/*.test.ts');
for (const file of files) {
  let content = fs.readFileSync(file, 'utf8');
  let changed = false;
  
  if (content.includes('getNpcPackageByPlotNpcId(') && !content.includes('getPlotNpcNumericId')) {
    content = content.replace(/import \{.*getNpcPackageByPlotNpcId.*\} from '\.\.\/src\/data\/npc_packages';/, (match) => {
      return match.replace('{', '{ getPlotNpcNumericId,');
    });
  }

  content = content.replace(/getNpcPackageByPlotNpcId\((['`][^'`]+['`])\)/g, "getNpcPackageByPlotNpcId(getPlotNpcNumericId($1)!)");
  // Also plotNpcName might need it?
  // Let's see `plotNpcName` definition: function plotNpcName(plotNpcId: string)
  // `plotNpcName` takes a string, then calls `getNpcPackageByPlotNpcId(plotNpcId)`!
  // So inside the tests that define `plotNpcName`, we need to change `plotNpcId: string` to `number` or wrap it.
  
  if (content !== fs.readFileSync(file, 'utf8')) {
    fs.writeFileSync(file, content);
    console.log('Fixed', file);
  }
}

File: fix_get_package.py
import os
import glob
import re

files = glob.glob('tests/**/*.test.ts', recursive=True)
for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    orig = content
    
    # 1. Replace getNpcPackageByPlotNpcId('...') with getNpcPackageByPlotNpcId(getPlotNpcNumericId('...')!)
    content = re.sub(r"getNpcPackageByPlotNpcId\((['`][^'`]+['`])\)", r"getNpcPackageByPlotNpcId(getPlotNpcNumericId(\1)!)", content)
    
    # 2. Add import for getPlotNpcNumericId if needed
    if "getPlotNpcNumericId" in content and "getPlotNpcNumericId" not in orig:
        content = re.sub(r"(import\s*\{[^\}]*)(getNpcPackageByPlotNpcId)", r"\1getPlotNpcNumericId, \2", content)
        
    # Fix plotNpcName as well, which is defined locally in some tests: function plotNpcName(plotNpcId: string)
    # We should let the test pass string to plotNpcName, but inside plotNpcName, call getPlotNpcNumericId
    content = re.sub(r"function plotNpcName\(plotNpcId: string\): string \{\n\s*const pack = getNpcPackageByPlotNpcId\(plotNpcId\);", 
                     r"function plotNpcName(plotNpcId: string): string {\n  const pack = getNpcPackageByPlotNpcId(getPlotNpcNumericId(plotNpcId)!);", content)

    if content != orig:
        with open(file, 'w', encoding='utf-8') as f:
            f.write(content)
        print("Fixed", file)

